test/tests/functional/pbs_reservations.py

```diff
@@ -2310,3 +2310,177 @@ def test_resv_reconfirm_holding_partial_nodes(self):
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+        j1 = Job(attrs={'Resource_List.walltime': 100})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+
+        now = int(time.time())
+        # Submit a standing reservation to occur every other minute for a
+        # total count of 2
+        start = time.time() + 10
+        now = start + 50
+        start = int(start)
+        end = int(now)
+        rid = self.submit_reservation(user=TEST_USER, select='1:ncpus=1',
+                                      rrule='FREQ=HOURLY;INTERVAL=2;COUNT=2',
+                                      start=start, end=end)
+        rid_q = rid.split('.')[0]
+        exp_attr = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, exp_attr, id=rid, offset=5)
+        self.server.holdjob(jid1)
+        # qrerun the jobs
+        self.server.rerunjob(jobid=jid1)
+        self.server.expect(JOB, {'job_state': 'H'}, id=jid1)
+        # qmove the job to reservation queue
+        self.server.movejob(jid1, rid_q)
+        self.server.expect(JOB, {'job_state': 'H', 'queue': rid_q},
+                           id=jid1)
+        self.server.rlsjob(jid1, 'u')
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.delete(jid1)
+        j2 = Job(attrs={'Resource_List.walltime': 10})
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+
+    @skipOnCpuSet
+    def test_shared_exclusive_job_in_same_rsv_blade(self):
+        """
+        Test shared and exclusive job in reservation assigned
+        to same blade
+        """
+        vn_attrs = {ATTR_rescavail + '.ncpus': 4,
+                    'sharing': 'default_excl'}
+        self.server.create_vnodes("vnode1", vn_attrs, 6,
+                                  self.mom, fname="vnodedef1")
+        # Submit a advance reservation (R1)
+        now = int(time.time())
+        a = {'Resource_List.select': '3:ncpus=4',
+             'reserve_start': now + 10,
+             'reserve_end': now + 1000}
+        r = Reservation(attrs=a)
+        rid = self.server.submit(r)
+        rid_q = rid.split('.')[0]
+        a = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
+        self.server.expect(RESV, a, id=rid)
+        a = {'Resource_List.select': '1:ncpus=2',
+             'Resource_List.place': 'shared',
+             'queue': rid_q}
+        j = Job(attrs=a)
+        j.set_sleep_time(999)
+        jid = self.server.submit(j)
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, offset=3)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        self.server.status(JOB)
+        job_node = j.get_vnodes()[0]
+        self.server.status(RESV)
+        resv_nodes = self.server.reservations[rid].get_vnodes()
+        errmsg = 'job node does not contain resv node'
```

Comment:
shall we say "job is not run on the nodes assigned to the reservation"
But, why do we need to test this?